app/Platform/Services/SystemGamesPageService.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services;
+
+use App\Enums\Permissions;
+use App\Models\System;
+use Illuminate\Http\Request;
+
+class SystemGamesPageService
+{
+    public function __construct(
+        protected GameListService $gameListService,
+    ) {
+    }
+
+    public function buildViewData(Request $request, System $system): array
```

Comment:
You could argue that this should just be part of `GameListService`, as that builds a large portion of the view data, then this just tweaks it a bit.

Reply:
That is indeed arguable - I do like the present separation of concerns though. It's conceivable `GameListService` could receive even more functionality in the future.